seisgo/noise.py

```diff
@@ -796,66 +960,124 @@ def stacking(corrdata,method='linear',par=None):
     # good to return
     return dstack,cc_time
 
-def rotation(bigstack,parameters,locs,flag):
-    '''
-    this function transfers the Green's tensor from a E-N-Z system into a R-T-Z one
+def get_locator(ds, sta):
+    inv = ds.waveforms[sta]['StationXML']
+    loc = []
+    id_list = inv.get_contents()['channels']
 
+    # Directly append unique location codes to `loc`
+    for id in id_list:
+        location = id.split('.')[2]
+        if location not in loc:  # Only append if it's not already in `loc`
+            loc.append(location)
+    
+    return [loc, inv]
+
+def rotation(sraw, rraw, channels=None):
```

Comment:
change the function name to be more specific, e.g., rotation_enz2rtz(). Or add an option on the rotation mode, mode='enz2rtz'. This will make the function expandable in the future when we need to rotate back from rtz2enz.

Reply:
I am not sure if the rotation matrix works for rtz2enz. But changing the name to rotation_enz2rtz() definitely works